sample-apps/donet-sample-app/Controllers/AppController.cs

```diff
@@ -0,0 +1,130 @@
+    {
+        private readonly AmazonS3Client s3Client = new AmazonS3Client();
+        private readonly HttpClient httpClient = new HttpClient();
+        private static Random rand = new Random(DateTime.Now.Millisecond);
+
+        public static readonly ActivitySource tracer = new(
+        "dotnet-sample-app");
+
+        public AppController() {}
+        
+        [HttpGet]
+        [Route("/outgoing-http-call")]
+        public string OutgoingHttp()
+        {
+            using var activity = tracer.StartActivity("outgoing-http-call");
+            activity?.SetTag("language", "dotnet");
+            activity?.SetTag("signal", "trace");
+            
+            var res = httpClient.GetAsync("https://aws.amazon.com/").Result;
+            string statusCode = res.StatusCode.ToString();
+            
+            // Request Based Metrics
+            Startup.metricEmitter.emitReturnTimeMetric(MimicLatency());
+            int loadSize = MimicPayLoadSize();
+            Startup.metricEmitter.apiRequestSentMetric();
+            Startup.metricEmitter.updateTotalBytesSentMetric(loadSize);
+            
+            return GetTraceId();
+        }
+
+        [HttpGet]
+        [Route("/aws-sdk-call")]
+        public string AWSSDKCall()
+        {
+            using var activity = tracer.StartActivity("aws-sdk-call");
+            activity?.SetTag("language", "dotnet");
+            activity?.SetTag("signal", "trace");
+
+            var res = s3Client.ListBucketsAsync().Result;
+            string statusCode = res.HttpStatusCode.ToString();
+            
+            // Request Based Metrics
+            Startup.metricEmitter.emitReturnTimeMetric(MimicLatency());
+            int loadSize = MimicPayLoadSize();
+            Startup.metricEmitter.apiRequestSentMetric();
+            Startup.metricEmitter.updateTotalBytesSentMetric(loadSize);
+            Startup.metricEmitter.totalTimeSentObserver.Add(3);
+
+            return GetTraceId();
+        }
+
+        [HttpGet]
+        [Route("/")]
+        public string Default()
+        {
+            return "Application started!";
+        }
+
+        [HttpGet]
+        [Route("/outgoing-sampleapp")]
+        public string OutgoingSampleApp()
+        {
+            using var activity = tracer.StartActivity("outgoing-sampleapp");
+            activity?.SetTag("language", "dotnet");
+            activity?.SetTag("signal", "trace");
+            string statusCode = "";
+
+            if (Program.cfg.SampleAppPorts.Length == 0) {
+                var res = httpClient.GetAsync("https://aws.amazon.com/").Result;
+                statusCode = res.StatusCode.ToString();
+            }
+            else {
+                foreach (string port in Program.cfg.SampleAppPorts) {
+                    if (!String.IsNullOrEmpty(port)) {
+                        Console.WriteLine(port);
+                        string uri = "http://127.0.0.1:" + port + "/outgoing-sampleapp";
```

Comment:
Does C#/ASP provide a better way to build URIs other than using string concatenation like this?

Reply:
I now use `string.Format()` for the uri in the latest commit